From 7b9532197fcca48b4b931c70166882aabe765975 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Wed, 8 May 2024 21:43:08 +0200
Subject: [PATCH] Fix issues with fenv

---
diff --git a/libstdc++-v3/include/c_compatibility/fenv.h b/libstdc++-v3/include/c_compatibility/fenv.h
index 6800814bc..f2d400991 100644
--- a/libstdc++-v3/include/c_compatibility/fenv.h
+++ b/libstdc++-v3/include/c_compatibility/fenv.h
@@ -32,13 +32,13 @@
 #pragma GCC system_header
 
 #include <bits/c++config.h>
-#if _GLIBCXX_HAVE_FENV_H
+#if !defined(__vax__) && _GLIBCXX_HAVE_FENV_H
 # include_next <fenv.h>
 #endif
 
 #if __cplusplus >= 201103L
 
-#if _GLIBCXX_USE_C99_FENV
+#if !defined(__vax__) && _GLIBCXX_USE_C99_FENV
 
 #undef feclearexcept
 #undef fegetexceptflag
diff --git a/libstdc++-v3/include/c_global/cfenv b/libstdc++-v3/include/c_global/cfenv
index fd32daa8d..22d97f047 100644
--- a/libstdc++-v3/include/c_global/cfenv
+++ b/libstdc++-v3/include/c_global/cfenv
@@ -37,11 +37,11 @@
 
 #include <bits/c++config.h>
 
-#if _GLIBCXX_HAVE_FENV_H
+#if !defined(__vax__) && _GLIBCXX_HAVE_FENV_H
 # include <fenv.h>
 #endif
 
-#ifdef _GLIBCXX_USE_C99_FENV
+#if !defined(__vax__) && defined(_GLIBCXX_USE_C99_FENV)
 
 #undef feclearexcept
 #undef fegetexceptflag
diff --git a/libstdc++-v3/src/c++17/floating_to_chars.cc b/libstdc++-v3/src/c++17/floating_to_chars.cc
index 2c9da977c..1b066b0c7 100644
--- a/libstdc++-v3/src/c++17/floating_to_chars.cc
+++ b/libstdc++-v3/src/c++17/floating_to_chars.cc
@@ -1048,7 +1048,7 @@ namespace
   {
     int len;
 
-#if _GLIBCXX_USE_C99_FENV_TR1 && defined(FE_TONEAREST)
+#if !defined(__sh__) && _GLIBCXX_USE_C99_FENV_TR1 && defined(FE_TONEAREST)
     const int saved_rounding_mode = fegetround();
     if (saved_rounding_mode != FE_TONEAREST)
       fesetround(FE_TONEAREST); // We want round-to-nearest behavior.
@@ -1082,7 +1082,7 @@ namespace
 #endif
     len = sprintf(buffer, format_string, args..., value);
 
-#if _GLIBCXX_USE_C99_FENV_TR1 && defined(FE_TONEAREST)
+#if !defined(__sh__) && _GLIBCXX_USE_C99_FENV_TR1 && defined(FE_TONEAREST)
     if (saved_rounding_mode != FE_TONEAREST)
       fesetround(saved_rounding_mode);
 #endif
-- 
2.36.6

